test/target.spec.ts

```diff
@@ -67,6 +67,27 @@ describe('Target', function () {
     ).toBe('Hello world');
     expect(await originalPage.$('body')).toBeTruthy();
   });
+  itFailsFirefox('should be able to use async waitForTarget', async () => {
+    const { page, server, context } = getTestState();
+
+    const [otherPage] = await Promise.all([
+      context
+        .waitForTarget((target) =>
+          target
+            .page()
+            .then(
+              (page) =>
+                page.url() === server.CROSS_PROCESS_PREFIX + '/empty.html'
+            )
+        )
+        .then((target) => target.page()),
+      page.evaluate(
+        (url: string) => window.open(url),
+        server.CROSS_PROCESS_PREFIX + '/empty.html'
+      ),
+    ]);
+    expect(otherPage.url()).toContain(server.CROSS_PROCESS_PREFIX);
```

Comment:
should we check for an exact URL match? Also I wonder if the `page` has the same URL as the `otherPage`? Can we make sure that we are waiting for the right target and they have distinct URLs?